corgi/tasks/sca.py

```diff
@@ -140,15 +139,11 @@ def _get_distgit_sources(source_url: str, package_type: str) -> list[Path]:
 
 def _archive_source(source_url: str, package_type: str) -> Tuple[Optional[Path], str]:
     # (scheme, netloc, path, parameters, query, fragment)
-    try:
-        url = urllib.parse.urlparse(source_url)
-    except ValueError as e:
-        logger.error("Invalid url passed to archive_source %s", e)
-        return None, ""
+    url = urlparse(source_url)
 
-    if url.scheme != "git":
-        logger.error("Cannot download raw source for anything but git protocol")
-        return None, ""
+    # We only support git, git+https, git+ssh
+    if not url.scheme.startswith("git"):
+        raise ValueError("Cannot download raw source for anything but git protocol")
```

Comment:
@jasinner I made this change here before we discussed changing the git-archive logic in standup today. Please let me know if I should drop this, or if only "git" should be explicitly supported, and not tunneled git protocols like git+https.

Reply:
I think it's fine to add it for now. I'll do a bit of investigation into what protocols are supported using the new git clone method in CORGI-158